src/ExtEventLoop.php

```diff
@@ -25,13 +25,32 @@ class ExtEventLoop implements LoopInterface
     private $readListeners = [];
     private $writeListeners = [];
     private $running;
+    private $signals;
+    private $signalEvents = [];
 
     public function __construct(EventBaseConfig $config = null)
     {
         $this->eventBase = new EventBase($config);
         $this->futureTickQueue = new FutureTickQueue();
         $this->timerEvents = new SplObjectStorage();
 
+        $this->signals = new SignalsHandler(
+            function ($signal) {
+                $this->signalEvents[$signal] = Event::signal($this->eventBase, $signal, function () use ($signal) {
+                    $this->signals->call($signal);
+                });
+                $this->signalEvents[$signal]->add();
+            },
+            function ($signal) {
+                $this->futureTick(function () use ($signal) {
```

Comment:
In come loops removing the signal handled within the function handling a signal will make it fail hard. Another way would be to use future tick in the `SignalHandler` when receiving the signal but that means the `SignalHandler` becomes aware of the event loop.

Reply:
Not sure I follow, can you provide a gist that shows this problem? Reading the documentation of our event loop implementations I could not find anything that supports this claim.

Reply:
Hmm odd can't reproduce is niet :zipper_mouth_face: . I'll remove the ticks

Reply:
And I'll add them again because this is what happens: https://travis-ci.org/reactphp/event-loop/jobs/290513376#L1401

Reply:
Interesting find! Looks like this was actually bug in PHP (https://bugs.php.net/bug.php?id=62452) that was fixed with PHP 7. Removing this in the next tick has some consequences, so I wonder if this can be worked around by keeping a reference to the closure instead (see also the PHP bug for some workarounds).

Reply:
@clue yeah also know why I couldn't see it locally, didn't ran the tests on 5.6 locally :zipper_mouth_face: . Just committed a possible fix, since I can't test locally I'm going to run them on Travis and squash the commits once fixed if it needs more then one try.

Reply:
@clue resolved the issue :tada: